src/app/signup/page.tsx

```diff
@@ -0,0 +1,237 @@
+"use client";
+
+import Image from "next/image";
+import Link from "next/link";
+import { useRecoilState } from "recoil";
+import { authState, AuthState, userTypeState } from "../recoil/atoms";
+import "../signin/page.css";
+import "./page.css";
+
+export default function SignIn() {
+  const [auth, setAuth] = useRecoilState<AuthState>(authState);
```

Comment:
What the reason why you’re using the same authState for sign up and sign in page? 
If I write all inputs to sign up page, but didn’t submit and go to sing in page. I already have inside email and password that I provided on sign up page earier.  And the same behaviour is in another way.
It’s because you’re using the same recoil state for both forms which is shared between all pages and the state is not reset after you change the page.

So in this case I see 2 possible solutions:
1) We need to create another state for sign up form. ( in this case it’s even better to have 2 separate React.useState on every page)
2) We need to make sure to clean all fields for authState when user leave the page.


<img width="1039" alt="Screenshot 2023-11-15 at 14 58 58" src="https://github.com/LorenaCapraru/song-sieve/assets/91835307/5862e6ff-8ea1-4877-82a1-519307723469">
<img width="1008" alt="Screenshot 2023-11-15 at 14 58 49" src="https://github.com/LorenaCapraru/song-sieve/assets/91835307/19f6965a-9174-463e-8b1f-dd59c6320bbd">

Reply:
Great catch, sure I have noticed this in test process.

Also will make sure to reset input values each time.